x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/v2/TransportRollupAction.java

```diff
@@ -427,42 +432,42 @@ public static String createRollupIndexMapping(
+                // the same rules with resize apply
+                continue;
+            }
+            // do not override settings that have already been set in the rollup index
+            if (targetSettings.keys().contains(key)) {
+                continue;
+            }
+            targetSettings.copy(key, sourceIndexMetadata.getSettings());
+        }
 
         /*
          * Add the source index name and UUID to the rollup index metadata.
          * If the source index is a rollup index, we will add the name and UUID
          * of the first index that we initially rolled up.
          */
-        String originalIndexName = IndexMetadata.INDEX_ROLLUP_SOURCE_NAME.exists(sourceIndexMetadata.getSettings())
-            ? IndexMetadata.INDEX_ROLLUP_SOURCE_NAME.get(sourceIndexMetadata.getSettings())
-            : sourceIndexName;
-        String originalIndexUuid = IndexMetadata.INDEX_ROLLUP_SOURCE_UUID.exists(sourceIndexMetadata.getSettings())
-            ? IndexMetadata.INDEX_ROLLUP_SOURCE_UUID.get(sourceIndexMetadata.getSettings())
-            : sourceIndexMetadata.getIndexUUID();
-
-        // Copy time series index settings from original index
-        List<String> indexRoutingPath = sourceIndexMetadata.getRoutingPaths();
-        Instant startTime = IndexSettings.TIME_SERIES_START_TIME.get(sourceIndexMetadata.getSettings());
-        Instant endTime = IndexSettings.TIME_SERIES_END_TIME.get(sourceIndexMetadata.getSettings());
-        IndexMode indexMode = IndexSettings.MODE.get(sourceIndexMetadata.getSettings());
-
-        return IndexMetadata.builder(rollupIndexMetadata)
-            .settings(
-                Settings.builder()
-                    .put(rollupIndexMetadata.getSettings())
-                    .put(IndexMetadata.INDEX_ROLLUP_SOURCE_NAME.getKey(), originalIndexName)
-                    .put(IndexMetadata.INDEX_ROLLUP_SOURCE_UUID.getKey(), originalIndexUuid)
-                    .put(IndexMetadata.INDEX_HIDDEN_SETTING.getKey(), sourceIndexMetadata.isHidden())
-                    // Add the time series index settings
-                    .put(IndexSettings.MODE.getKey(), indexMode)
-                    .putList(IndexMetadata.INDEX_ROUTING_PATH.getKey(), indexRoutingPath)
-                    .put(IndexSettings.TIME_SERIES_START_TIME.getKey(), startTime.toString())
-                    .put(IndexSettings.TIME_SERIES_END_TIME.getKey(), endTime.toString())
-            );
+        if (IndexMetadata.INDEX_ROLLUP_SOURCE_UUID.exists(sourceIndexMetadata.getSettings()) == false
+            || IndexMetadata.INDEX_ROLLUP_SOURCE_UUID.exists(sourceIndexMetadata.getSettings()) == false) {
+            Index sourceIndex = sourceIndexMetadata.getIndex();
+            targetSettings.put(IndexMetadata.INDEX_ROLLUP_SOURCE_NAME.getKey(), sourceIndex.getName())
+                .put(IndexMetadata.INDEX_ROLLUP_SOURCE_UUID.getKey(), sourceIndex.getUUID());
```

Comment:
Do we already have a test for doing a rollup-of-a-rollup where the original index is maintained in the rollup source/uuid? If not, could you add one?

Reply:
Right now rollup-of-rollups does not fully work. I plan to add support for rollup-of-rollup in a following PR. 
I will definitely add tests in that PR

This part is only plumbing that is going to be needed for rollups of rollups

Reply:
> Right now rollup-of-rollups does not fully work.

As some subsequent work (not in this PR), does it make sense to reject the request if someone tries to rollup an index that's already been rolled up?

Reply:
In data streams this cannot happen because no two indices for the same period can co-exist.

When it comes to concrete indices, there are very valid use cases for allowing multiple rollup indices of the same source index. For example having multiple rollup intervals (hourlies, dailies etc) or having multiple timezones (right now we support only UTC, but tin the future we should support more timezones).